client/my-sites/checkout/composite-checkout/lib/cart-local-storage.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import type { RequestCartProduct } from '@automattic/shopping-cart';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:cart-local-storage' );
+
+let cartProductsFallbackStorage: Partial< RequestCartProduct >[] = [];
+
+const localStorageKey = 'automatticShoppingCartProducts';
+
+// Used by signup; see https://github.com/Automattic/wp-calypso/pull/44206
+// These products are likely missing product_id.
+export function getCartFromLocalStorage(): Partial< RequestCartProduct >[] {
+	try {
+		const products = JSON.parse( window.localStorage.getItem( localStorageKey ) || '[]' );
+		if ( Array.isArray( products ) ) {
+			return products;
+		}
+	} catch ( err ) {
+		debug( 'An error ocurred reading saved cart products from localStorage: ' + err );
+	}
+	return cartProductsFallbackStorage;
+}
+
+export function addCartItemsToLocalStorage( newItems: Partial< RequestCartProduct >[] ): void {
```

Comment:
This function appends new items to existing items, which is why I used the word `add`. I think that `update` would imply that you could somehow use it to change what's currently in localStorage (which is what `saveCartItemsToLocalStorage` does).

Reply:
It makes sense to keep the name since we're explicitly using the function to append new items but the spread operator will update existing items as well!